onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1031,7 +1017,29 @@ Applies to session load, initialization, etc. Default is 0.)pbdoc")
           },
           R"pbdoc(Graph optimization level for this session.)pbdoc")
       .def_readwrite("use_deterministic_compute", &SessionOptions::use_deterministic_compute,
-                     R"pbdoc(Whether to use deterministic compute. Default is false.)pbdoc");
+                     R"pbdoc(Whether to use deterministic compute. Default is false.)pbdoc")
+      .def(
+          "add_free_dimension_override",
+          [](SessionOptions* options, const char* dim_name, int64_t dim_value)
+              -> void { options->free_dimension_overrides.push_back(
+                            onnxruntime::FreeDimensionOverride{
+                                dim_name,
+                                onnxruntime::FreeDimensionOverrideType::Denotation,
+                                dim_value}); },
+          "Rpbdoc(Specify the dimension size for each denotation associated with an input's free dimension.)pbdoc")
+      .def(
+          "add_free_dimension_override_by_name",
+          [](SessionOptions* options, const char* dim_name, int64_t dim_value)
+              -> void { options->free_dimension_overrides.push_back(
+                            onnxruntime::FreeDimensionOverride{
+                                dim_name,
+                                onnxruntime::FreeDimensionOverrideType::Name,
+                                dim_value}); },
+          "Rpbdoc(Specify values of named dimensions within model inputs.)pbdoc")
+      .def(
+          "clone", [](SessionOptions* options) -> SessionOptions { 
```

Comment:
Is there a more [Pythonic way](https://docs.python.org/3/library/copy.html#copy.deepcopy) to achieve this?

Reply:
We could. But AFAIK we don't expose a `SessionOptions` class in Python directly (where we could expose a clone() that does a deep copy). Hence, putting this in the binding class for ease. Am I overlooking something ?